voidcrew/code/modules/mining/machine_bluespaceminer.dm

```diff
@@ -0,0 +1,117 @@
+	if(in_range(user, src) || isobserver(user))
+		. += "<span class='notice'>A small screen on the machine reads, \"Efficiency at [multiplier * 100]%\"</span>"
+		. += "<span class='notice'>A small screen on the machine reads, \"Power Usage at [idle_power_usage / 1000]kWs\"</span>"
+		if(multiplier >= 5)
+			. += "<span class='notice'>Bluespace generation is active.</span>"
+
+/obj/machinery/power/bluespace_miner/RefreshParts()
+	multiplier = 1
+	for(var/obj/item/stock_parts/micro_laser/laser_tier in component_parts)
+		multiplier += (laser_tier.rating-1)/3
+	for(var/obj/item/stock_parts/capacitor/capacitor_tier in component_parts)
+		power_coeff += (capacitor_tier.rating-1)*0.55
+		multiplier += (capacitor_tier.rating-1)*1.3
+	idle_power_usage = base_power_usage * power_coeff
+
+
+/obj/machinery/power/bluespace_miner/Destroy()
+	materials = null
+	return ..()
+
+
+/obj/machinery/power/bluespace_miner/examine(mob/user)
+	. = ..()
+	if(!materials?.silo)
+		. += "<span class='notice'>No ore silo connected. Use a multi-tool to link an ore silo to this machine.</span>"
+	else if(materials?.on_hold())
+		. += "<span class='warning'>Ore silo access is on hold, please contact the quartermaster.</span>"
+
+/obj/machinery/power/bluespace_miner/process()
+	update_icon_state()
+	if(!materials?.silo || materials?.on_hold())
+		return
+	var/datum/component/material_container/mat_container = materials.mat_container
+	if(!mat_container || panel_open || !powered())
+		return
+	/// The ore that its going to generate and insert into the Ore Silo
+	var/datum/material/ore = pick(ore_rates)
+	mat_container.bsm_insert(((ore_rates[ore] * 1000) * multiplier), ore)
+
+/**Used whenever the Bluespace Miner needs to insert its materials into an orm
+  * amt -> Amount of resource being deposited
+  * datum/material/mat -> The material being deposited
+  */
+/datum/component/material_container/proc/bsm_insert(amt, datum/material/mat)
+	if(!istype(mat))
+		mat = SSmaterials.GetMaterialRef(mat)
+	if(amt > 0 && has_space(amt))
+		/// Tracks the total amount of ore deposited in this bsm_insert
+		var/total_amount_saved = total_amount
+		if(mat)
+			materials[mat] += amt
+			total_amount += amt
+		else
+			/// Amount to input
+			for(var/input in materials)
+				materials[input] += amt
+				total_amount += amt
+		return (total_amount - total_amount_saved)
+	return FALSE
+
+/obj/machinery/power/bluespace_miner/update_icon_state()
+	if(!powered())
+		if(!panel_open)
+			icon_state = "bsminer-unpowered"
+		else
+			icon_state = "bsminer-unpowered-maintenance"
+	else
+		if(!panel_open)
+			icon_state = "bsminer"
+		else
+			icon_state = "bsminer-maintenance"
+
+/obj/machinery/power/bluespace_miner/crowbar_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_deconstruction_crowbar(tool, FALSE))
+		return TRUE
+
+/obj/machinery/power/bluespace_miner/screwdriver_act(mob/living/user, obj/item/tool)
+	. = TRUE
+	if(..())
+		return
+	if(!tool.tool_behaviour == TOOL_SCREWDRIVER)
```

Comment:
This is pointless, `screwdriver_act` will only call if the tool is a screwdriver anyways